src/application/engine/Menu.java

```diff
@@ -1,4 +1,73 @@
 package application.engine;
 
+import application.coffe.Coffee;
+
+import java.util.Map;
+import java.util.Scanner;
+
 public class Menu {
+    private final Scanner scanner;
+    private final Machine machine;
+    private final Map<String, Coffee> recipeBook;
+
+    public Menu(Scanner scanner, Machine machine, Map<String, Coffee> recipeBook) {
+        this.scanner = scanner;
+        this.machine = machine;
+        this.recipeBook = recipeBook;
+    }
+
+    public void run() {
+        String result = "";
+        boolean control = true;
+        while (control) {
+            System.out.println("Write action (buy, fill, take, remaining, exit):");
+            var response = scanner.next();
+            switch (response) {
+                case "buy" -> System.out.println(makingCoffee(result));
+                case "fill" -> makeFill();
+                case "take" -> System.out.println(machine.take());
+                case "remaining" -> System.out.println(machine.status());
+                default -> System.out.println("No such command");
+                case "exit" -> control = false;
+            }
+        }
+    }
+
+    private String makingCoffee(String result) {
+        System.out.println("What do you want to buy? " + parseRicipe() + "back - to main menu:");
+        String number = scanner.next();
+        switch (number) {
+            case "back":
+                break;
+            default:
+                if (recipeBook.containsKey(number)) {
+                    result = machine.makeCoffee(recipeBook.get(number));
+                } else {
+                    result = "There no such type of coffee";
+                }
+                break;
+        }
+        return result;
+    }
+
+    private String parseRicipe() {
+        StringBuilder builder = new StringBuilder();
+        for (String s : recipeBook.keySet()) {
```

Comment:
It is a valid way to iterate through Map. But when we need key and value it is more convenient to use one of the following approach:
1)
for (var entry : map.entrySet()) {
       System.out.println(entry.getKey() + "/" + entry.getValue());
}
2) 
for (Map.Entry<String, String> entry : map.entrySet()) {
    System.out.println(entry.getKey() + "/" + entry.getValue());
}
3)
map.forEach((key, value) -> {
        System.out.println(key+" : "+ value);
    });